packages/neos-ui-editors/src/SecondaryEditors/ImageCropper/index.js

```diff
@@ -159,8 +159,14 @@ export default class ImageCropper extends PureComponent {
             const normalizedAspectRatioHeight = currentAspectRatioStrategy.height / aspectRatioGcd;
 
             // pixel perfect calculations
-            const naturalCropWidth = Math.floor(imageWidth * (cropArea.width / 100) / normalizedAspectRatioWidth) * normalizedAspectRatioWidth;
-            const naturalCropHeight = naturalCropWidth / normalizedAspectRatioWidth * normalizedAspectRatioHeight;
+            let naturalCropWidth = Math.floor(imageWidth * (cropArea.width / 100) / normalizedAspectRatioWidth) * normalizedAspectRatioWidth;
+            let naturalCropHeight = naturalCropWidth / normalizedAspectRatioWidth * normalizedAspectRatioHeight;
+
+            while (naturalCropHeight > imageHeight) {
+                // can't crop area larger than image itself, so keep subtracting normalized aspect ratio values until area is valid
+                naturalCropHeight -= normalizedAspectRatioHeight;
+                naturalCropWidth -= normalizedAspectRatioWidth;
+            }
```

Comment:
The assumption being made in the following statements is that imageHeight will be positive or 0.

The loop then terminates when one of the following conditions is met:

`normalizedAspectRatioHeight > 0 ` or `naturalCropHeight < 0`

If we actually substitute the variables in naturalCropHeight variable declaration and ignore the floor function since it wouldn't change the sign , you eventually get a minimized form like

`let naturalCropHeightWithoutFloor = cropArea.width * imageWidth * (currentAspectRatioStrategy.height /( 100 * currentAspectRatioStrategy.width))` (see [Wolframalpha](https://www.wolframalpha.com/input?i=%28%28a+*+%28b+%2F+100%29+%2F+%28c+%2F+e%29%29+*+%28c+%2F+e%29%29+%2F+%28c+%2F+e%29+*+%28f%2Fe%29+%3D+b+*+a+*+f+%2F+%28100+*+c%29))

From that we can see that if either passed variable in crop configuration is negative, naturalCropHeight will be smaller than zero and the loop will never start. If both configuration dimension variables are positive the `aspectRatioGcd` will be positive and therefore `normalizedAspectRatioHeight` will be positive, hence the loop with exit eventually.